pump/enum_api_pin.go

```diff
@@ -68,14 +67,19 @@ func (a *APIPinEnumerator) directCIDs(out chan<- BlockInfo) error {
 func (a *APIPinEnumerator) streamCIDs(out chan<- BlockInfo) error {
 	s := shell.NewShell(a.URL)
 
-	pins, err := pinsStream(s)
+	pinStream, err := s.PinsStream(context.Background())
 	if err != nil {
 		return err
 	}
 
+	// Reset the total count because if we would start doing a.totalCount++ from -1,
+	// we end up with overflowing progress bar like: `QmU... 0s  10 / 9 [========]`
+	// with this adjustment we get expected: `QmU... 0s  10 / 10 [========]`
+	a.totalCount = 0
```

Comment:
```suggestion
	// Now that we started the query we can properly count, which means we need to
	// reset the counter to zero instead of -1 (meaning unknown).
	a.totalCount = 0
```